File: android/app/src/main/kotlin/app/hapi/companion/feature/newsession/NewSessionCatalogs.kt
package app.hapi.companion.feature.newsession

/** A `(wire value, display label)` pair for the option pickers. */
data class OptionItem(val value: String, val label: String)

/**
 * Static option catalogs for the create form — data ports of
 * `shared/src/models.ts` (`CLAUDE_MODEL_LABELS`), `shared/src/effort.ts`
 * (`CLAUDE_EFFORT_LABELS`) and the web `CODEX_REASONING_EFFORT_OPTIONS`
 * (`web/src/components/NewSession/types.ts`). Kept in the feature (not
 * `:core:protocol`'s catalog package) because they are create-form option
 * lists, not wire-format contracts.
 */
object NewSessionCatalogs {

    /** `'auto'` sentinel rows use the web's "Default" label. */
    val CLAUDE_MODELS: List<OptionItem> = listOf(
        OptionItem("auto", "Default"),
        OptionItem("sonnet", "Sonnet"),
        OptionItem("sonnet[1m]", "Sonnet 1M"),
        OptionItem("opus", "Opus"),
        OptionItem("opus[1m]", "Opus 1M"),
        OptionItem("fable", "Fable"),
        OptionItem("fable[1m]", "Fable 1M"),
    )

    val CLAUDE_EFFORTS: List<OptionItem> = listOf(
        OptionItem("auto", "Auto"),
        OptionItem("low", "Low"),
        OptionItem("medium", "Medium"),
        OptionItem("high", "High"),
        OptionItem("xhigh", "XHigh"),
        OptionItem("max", "Max"),
    )

    /** Static codex fallback when the model row advertises no efforts (web drops `max` for codex). */
    val CODEX_REASONING_EFFORTS: List<OptionItem> = listOf(
        OptionItem("default", "Default"),
        OptionItem("low", "Low"),
        OptionItem("medium", "Medium"),
        OptionItem("high", "High"),
        OptionItem("xhigh", "XHigh"),
    )

    /** Capitalized label for a server-advertised effort id (`high` → `High`). */
    fun effortLabel(value: String): String =
        value.replaceFirstChar { if (it.isLowerCase()) it.titlecase() else it.toString() }
}

File: ios/Packages/HapiKit/Sources/HapiProtocol/Catalog/NewSessionCatalogs.swift
import Foundation

/// A `(wire value, display label)` pair for the create-form option pickers.
public struct NewSessionOption: Equatable, Hashable, Sendable {
    public let value: String
    public let label: String

    public init(value: String, label: String) {
        self.value = value
        self.label = label
    }
}

/// Static option catalogs for the new-session form. The claude rows are
/// **derived** from `ClaudeModels` / `ClaudeEfforts` (issue #39: one source
/// of truth shared with `ModelCatalog`'s composer builders) rather than
/// duplicating the preset/label tables. Only the codex reasoning list is its
/// own data — a port of the web `CODEX_REASONING_EFFORT_OPTIONS`
/// (`web/src/components/NewSession/types.ts`; the web drops `max` for codex
/// — `EffortField.tsx`). Android keeps a feature-local copy
/// (`NewSessionCatalogs.kt`); here the catalogs live in one package so the
/// pure form logic and the app UI share one source.
public enum NewSessionCatalogs {
    /// `'auto'` sentinel row uses the web's "Default" label; preset rows come
    /// from `ClaudeModels` in picker order.
    public static let claudeModels: [NewSessionOption] =
        [NewSessionOption(value: "auto", label: "Default")]
            + ClaudeModels.presets.map {
                NewSessionOption(value: $0, label: ClaudeModels.label(for: $0) ?? $0)
            }

    /// `'auto'` sentinel row, then the `ClaudeEfforts` levels in ascending
    /// order.
    public static let claudeEfforts: [NewSessionOption] =
        [NewSessionOption(value: "auto", label: "Auto")]
            + ClaudeEfforts.levels.map {
                NewSessionOption(value: $0, label: ClaudeEfforts.label(for: $0) ?? effortLabel($0))
            }

    /// Static codex fallback when the model row advertises no efforts
    /// (the web drops `max` for codex — `EffortField.tsx`).
    public static let codexReasoningEfforts: [NewSessionOption] = [
        NewSessionOption(value: "default", label: "Default"),
        NewSessionOption(value: "low", label: "Low"),
        NewSessionOption(value: "medium", label: "Medium"),
        NewSessionOption(value: "high", label: "High"),
        NewSessionOption(value: "xhigh", label: "XHigh"),
    ]

    /// Capitalized label for a server-advertised effort id (`high` → `High`).
    public static func effortLabel(_ value: String) -> String {
        ModelCatalog.capitalizedFirst(value)
    }
}
